frontend/src/app/data-entry/episode-form/episode-entities-form/episode-entities-form.component.ts

```diff
@@ -0,0 +1,160 @@
+import { Component, ElementRef, Input, OnChanges, OnDestroy, SimpleChanges, ViewChild } from "@angular/core";
+import { map, Observable, Observer, Subject, switchMap, tap, withLatestFrom } from "rxjs";
+import { entityTypeNames, formStatusSubject } from "../../shared/utils";
+import { actionIcons } from "@shared/icons";
+import { MutationResult } from "apollo-angular";
+import { FormService } from "../../shared/form.service";
+import {
+    CreateEpisodeEntityLinkInput,
+    DataEntryCreateEpisodeEntityLinkGQL,
+    DataEntryDeleteEpisodeEntityLinkGQL,
+    DataEntryDeleteEpisodeEntityLinkMutationVariables,
+    DataEntryEpisodeEntitiesGQL,
+    DataEntryEpisodeEntitiesQuery,
+    Entity
+} from "generated/graphql";
+import { takeUntilDestroyed } from "@angular/core/rxjs-interop";
+import { splat, differenceBy } from "@shared/utils";
+
+type EntityPropertyName = 'agents' | 'gifts' | 'letters' | 'spaces';
+
+const REFETCH_QUERIES = ['DataEntryEpisodeEntities'];
+
+let nextID = 0;
+
+@Component({
+    selector: "lc-episode-entities-form",
+    templateUrl: "./episode-entities-form.component.html",
+    styleUrls: ["./episode-entities-form.component.scss"],
+})
+export class EpisodeEntitiesFormComponent implements OnChanges, OnDestroy {
+    @Input() entityType: Entity = Entity.Agent;
+
+    data$: Observable<DataEntryEpisodeEntitiesQuery>;
+
+    linkedEntities$: Observable<{ id: string, name: string }[]>;
+    availableEntities$: Observable<{ name: string, id: string }[]>;
+
+    addEntity$ = new Subject<string>();
+    removeEntity$ = new Subject<string>();
+    actionIcons = actionIcons;
+    status$ = formStatusSubject();
+    id = `episode-entities-${nextID++}`;
+
+    private mutationObserver: Partial<Observer<MutationResult>> = {
```

Comment:
This confused me because of the [Web API with the same name](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver).